pubtools/_quay/quay_steps.py

```diff
@@ -0,0 +1,671 @@
+from __future__ import print_function
+import re
+import six
+import pkg_resources
+
+from .utils.stepper import Step, StepFailedError
+from .utils.logger import log_jsonl
+
+
+LOG_INDENT = "  - "
+
+
+class StepSanitizeContainerPushItems(Step):
+    """Filter push items with error state or with empty file_path and ignore non docker items.
+
+    Expected step args:
+        ()
+    Expected external resources:
+        "log_debug": log debug callback or None
+        "log_info": log info callback or None
+        "log_warning": log warning callback or None
+        "log_error": log error callback or None
+        "push_items": list of push items
+    Produced details:
+        {"item": str "state": (ready, not-container, error, no-pull-data, ok)}
+    Results:
+        list of indexes of container push items in push_items list
+    """
+
+    NAME = "StepSanitizeContainerPushItems"
+
+    def _init_details(self):
+        push_items = self.external_resources["push_items"]
+        self._details = []
+        for item in push_items:
+            self._details.append({"item": str(item), "state": "ready"})
+
+    def _update_details(self, details):
+        index, state = details
+        self._details[index]["state"] = state
+
+    @log_jsonl("SanitizeContainerPushItems")
+    def _run(self, on_update=None):
+        push_items = self.external_resources["push_items"]
+        log_info = self.external_resources.get("log_info", lambda *args, **kwargs: ())
+        log_warning = self.external_resources.get(
+            "log_warning", lambda *args, **kwargs: ()
+        )
+        log_error = self.external_resources.get("log_error", lambda *args, **kwargs: ())
+
+        container_push_items = []
+        _on_update = on_update if on_update else lambda: ()
+        failed = False
+        for i, item in enumerate(push_items):
+            log_info("Processing %s", str(item))
+            item_error = False
+            item_state = "ok"
+            if item.file_type != "docker_v2":
```

Comment:
File type "docker_v2" was removed from Pub. Use "docker" here?

Reply:
right, I forgot it was changed. Thanks for pointing this out